sentinelhub/data_sources.py

```diff
@@ -0,0 +1,522 @@
+"""
+Module defining data sources
+"""
+import re
+import warnings
+from enum import Enum, EnumMeta
+from typing import Tuple
+from dataclasses import dataclass, asdict
+
+from aenum import extend_enum
+
+from .config import SHConfig
+from .constants import ServiceUrl
+from .exceptions import SHDeprecationWarning
+
+
+class _Source:
+    """ Types of satellite sources
+    """
+    SENTINEL2 = 'Sentinel-2'
+    SENTINEL1 = 'Sentinel-1'
+    LANDSAT8 = 'Landsat 8'
+    MODIS = 'MODIS'
+    DEM = 'Mapzen DEM'
+    BYOC = 'BYOC'
+    BATCH = 'BATCH'
+    LANDSAT5 = 'Landsat 5'
+    LANDSAT7 = 'Landsat 7'
+    SENTINEL3 = 'Sentinel-3'
+    SENTINEL5P = 'Sentinel-5P'
+    ENVISAT_MERIS = 'Envisat Meris'
+
+
+class _SensorType:
+    """ Types of satellite sensors
```

Comment:
Types is not a correct word, and also it is a bit of a conundrum; MSI is a (satellite/sensor) type, but IW and EW are swath modes. I do not know what would be the "common denominator" for these.

I'll try to find a better name, cause for the moment I can't think of one.

Reply:
Good point. For Sentinel-1 I added here `C-SAR` and moved `IW` and `EW` to a new parameter swath mode. Now it makes much more sense.